### PR TITLE
Rule8897

### LINKED ISSUE #63: Rule 8897 : Parental or child factors at assessment information is missing from a completed assessment
Where present, if \<AssessmentAuthorisationDate\> (N00160) is on or after [Start_Of_Census_Year] then one or more \<AssessmentFactors\> (N00181) must be present within the same assessment module and must be a valid code

### PR DESCRIPTION
**Pull request by WendyWigan**
Closes #63

### CODE REVIEW

File: cin_validator/rules/cin2022_23/rule_8897.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+    # Where present, if <AssessmentAuthorisationDate> (N00160) is on or after [Start_Of_Census_Year] then one or more <AssessmentFactors> (N00181) must be present within the same assessment module and must be a valid code
+    # Get collection period
+    header = data_container[Header]
+    ref_date_series = header[ReferenceDate]
+    collection_start, collection_end = make_census_period(ref_date_series)
+    condition1 = (df[AssessmentAuthorisationDate] >= collection_start) & (
+        df[AssessmentAuthorisationDate].notna()
+    )
+    condition2 = df[AssessmentFactors].isna()
```
Comment (willLPD2I) [inline cin_validator/rules/cin2022_23/rule_8897.py:52]: Hi Wendy, this rule needs you to check both if AssessmentFactor is present AND if it's a valid code. The codes can be found here: https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/1025195/Children_in_need_census_2022_to_2023_guide.pdf

To test if particular column on a particular row contains a valid code, I'd first make a list of valid codes, then use the .isin(codes_list_name) method on the column of the df, just as you have used .isna() and .notna() for conditions 1 and 2. If you feel you don't have time to do this, I'm happy to do it and send you an email with pictures explaining what I've done.

File: cin_validator/rules/cin2022_23/rule_8897.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        row_df=df_issues,
+    )
+
+
+def test_validate():
+    # Create some sample data such that some values pass the validation and some fail.
+    fake_data = pd.DataFrame(
+        [
+            {
+                "LAchildID": "child1",
```
Comment (willLPD2I) [inline cin_validator/rules/cin2022_23/rule_8897.py:89]: As you need to check fro valid codes, you'll need to add some passing and failing codes here too!

File: cin_validator/rules/cin2022_23/rule_8897.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    # Create the dataframe which you expect, based on the fake data you created. It should have two columns.
+    # - The first column is ERROR_ID which contains the unique combination that identifies each error instance, which you decided on earlier.
+    # - The second column in ROW_ID which contains a list of index positions that belong to each error instance.
+
+    # The ROW ID values represent the index positions where you expect the sample data to fail the validation check.
+    expected_df = pd.DataFrame(
+        [
+            {
+                "ERROR_ID": (
+                    "child3",
```
Comment (willLPD2I) [inline cin_validator/rules/cin2022_23/rule_8897.py:159]: This'll also need to account for the codes

**Review by willLPD2I — CHANGES_REQUESTED** [review]

**Comment by WendyWigan** [comment]
Hi Will

I don’t think that I will be able to do this as plans have changed for this week. I’m happy for you to complete and send me screen shots of how it has been updated/completed.

Regards


Wendy Yates
Research & Information Officer
Joint Intelligence Unit (Mon-Thurs only)
Resources Directorate
Wigan Council
Wigan New Town Hall, Library Street, Wigan WN1 1YN
01942 489996 (89996) CURRENTLY NOT IN USE
www.wigan.gov.uk
www.twitter.com/wigancouncil
www.facebook.com/WiganCouncilOnline

**Comment by willLPD2I** [comment]
Great Wendy, I’ll do it!

William Levack-Payne (He/Him)
Data Advocate (Data to Insight)
Children’s Services

07732 824391 | ***@***.******@***.***>

**Review by willLPD2I — APPROVED** [review]

**Pull request merged**